Modify governance to clarify rules regarding votes and the majority required (#22)
neoforged/governance: voting.md

# Team votes

> To ensure that members of a team are in agreement regarding a topic under their purview, a vote may be called by any member of the team or by a member of the Steering Council.  

Reviewer: I think this should go in the core governance document as it defines how the teams exert their decision making power.

Reviewer 2: I thin core should be more about overview. And vote details be split out. The core document can have a sentence summary and link that points to here as well to keep core more cleaner while still having the relevant info discoverable

The votes take the form of Discord polls, sent to one of the following channels:
- `#project-talk` for votes concerning all teams of the Project;
- `#maintenance-talk` for votes concerning the Maintainers team;
- `#internal-private` for votes that need to be held privately or concerning internal team structures;
- `#moderation-private` for votes concerning the Moderators team.

> Any accompanying contextual information must be sent in messages directly below the poll, alongside a ping of the role of the team in question.

Reviewer: Directly below or directly above in practice, so this is too specific.
```suggestion
Any accompanying contextual information must be sent in messages next to the poll, alongside a ping of the role of the team in question.
```

Author: My line of thought was that it's easier to pin the poll and scroll down than up for further information.

Author: Updated it to "next to" but added a recommendation for sending them below so that it's easier to navigate to it when you get sent a link to the poll.

> Voting takes place in the 24 hours following the creation of the poll, unless the Steering Council agrees to lift waiting requirements for situations that require prompt resolutions. If this is the case, the Council must clearly explain the context that lead to the decision of shortening the vote's duration.

Reviewer: > unless the Steering Council agrees to lift waiting requirements for situations that require prompt resolutions

Where is this coming from? Was there a case in the past where this was needed? You can instead mention that the vote can be resolved early if a majority of the entire team size is reached (e.g. 12+ votes for yes already with a maintainer team size of 23)

Reviewer 2: The Steering Council part should be removed imo. And the 24 hour limit should be stated as a minimum for members to have a chance to see the poll. Longer polls can be done for non-time sensitive stuff that might end up with more discussions during voting. Majority reach for entire possible team ending vote early is hmm. People could change their mind. How about supermajority >66% can end vote early?

Author: I just removed it entirely, I don't think we'll have circumstances so critical that we have to hold a few hour votes, by then it is not a vote, rather, a consensus. If the need arises we can revisit it later.

> For a decision to be taken by vote, a *simple* majority of the votes cast is required - that is, more than half of the votes cast must be in favour of the final decision. If the vote ends in a tie (no option received more than half of the votes), a majority of the members of the Steering Council will cast a tie breaking vote.

Reviewer: ```suggestion
For a decision to be taken by vote, a *simple* majority of the votes cast is required - that is, strictly more than half of the votes cast must be in favour of the final decision. If the vote ends in a tie (no option strictly received more than half of the votes), a majority of the members of the Steering Council will cast a tie breaking vote.

```

Reviewer 2: The steering council part should be removed here. Steering Council would just be for calling up votes to be done that were missed. A tie would be the status quo option. Or if no status quo option, the poll maker will state ahead of time which option a tie will default to.

> - As an exception to the rule above, votes that seek to overrule a previous decision require a *qualified* majority - i.e., two thirds of the votes cast have to be in favour.

Reviewer: ```suggestion
- As an exception to the rule above, votes that seek to overrule a previous decision require a *qualified* majority - i.e., at least two thirds of the votes cast have to be in favour.

```

# Team votes
To ensure that members of a team are in agreement regarding a topic under their purview, a vote may be called by any member of the team or by a member of the Steering Council.  
The votes take the form of Discord polls, sent to one of the following channels:
- `#project-talk` for votes concerning all teams of the Project;
- `#maintenance-talk` for votes concerning the Maintainers team;
- `#internal-private` for votes that need to be held privately or concerning internal team structures;
- `#moderation-private` for votes concerning the Moderators team.

Any accompanying contextual information must be sent in messages next to the poll, alongside a ping of the role of the team in question. It is recommended that contextual information is sent directly below the poll, to increase discoverability.  
Any further discussion regarding the vote should be, to the greatest extent possible, contained in a dedicated thread.

> Voting takes place in the 24 hours following the creation of the poll. Votes that are not time-sensitive may have a longer voting period, but it may not exceed one week.

Reviewer 3: I find these time frames a bit short. Sometimes, I'm not online on Sundays and would thus miss a vote that gets started in the night from Saturday to Sunday. 48h for normal votes would be better, 24h only for really urgent matters, and the maximum time frame should be more like 2 weeks.

Author: 2 weeks is an extremely long voting period. Spreading any potential discussions over multiple weeks is detrimental as opinions expressed will get lost and forgotten. Additionally, if one is unable to vote within a week I think that they should not participate in the vote because they would be uninformed, as votes most times are the result of discussions that have been recently held. 
24 hours has been the time frame we have been using for the past 2 years and I would like to think that we have been good at determining whether more time is needed on a case-by-case basis.
Others can comment on this too but I don't agree.

Reviewer 4: If the maximum vote time is two weeks and the vote needs a second round (ie 3+ options and none reach majority), thats a whole month to wait for a decision to be reached, which is a very long time for most things.

Additionally there is no 48hr/2 day option in Discord polls, so 3 days would have to be the minimum in that case. This is too long imo, especially for moderator decisions with things like modmail tickets, where us going silent (as it appears to the user) for 3 days is probably not a good idea. Even 24 hours can be a long time in that case, but reducing the minimum time to less is a bad idea due to timezones (especially since the next option down in Discord polls is 8hrs).

Reviewer: 24h seems like a good default to me. It's what we have typically done already.

For a decision to be taken by vote, a *simple* majority of the votes cast is required - that is, strictly more than half of the votes cast must be in favour of the final decision.
- As an exception to the rule above, votes that seek to overrule a previous decision require a *qualified* majority - i.e., at least two thirds of the votes cast have to be in favour.

In cases involving multiple-choice polls, if no option does not meet the requirements set above, the two choices with the most votes will be put to a run-off vote.

> A vote that ends in a tie (i.e., the cast votes have been *perfectly* distributed to each option) or one in which no option has met the majority requirement is considered **failed**. In this case, the status quo will be preserved. Failed votes may be re-attempted after 30 days.

Reviewer 4: This doesn't cover multiple choice polls with a tie in the top 2 (ie 6-5-5-2-2). 
A simple 'any tied can also be included in the run-off' could work, and in the case of a tie that covers the entire poll (ie 6-5-5, or 6-2-2-2-2) it would be considered failed, although I don't know how to describe that.  However, this also doesn't consider something like 6-5-5-0, where the run-off would likely have the same results due to other options having 0 votes.
I hate edge cases.

Author: Good point, I suppose we should just keep doing runoffs? Though technically that can deadlock us. Hmn

> Decisions made by vote cannot be revisited after less than 90 days from their vote.

Reviewer 5: How do we deal with externalities that force our hand? Specifically: if the underlying assumptions change radically due to IT-Security discoveries, decisions by Mojang, decisions by Launchers, etc. our hands should not be tied for 90 days to revise a decision.

Reviewer 5: Alright, we'll just pass that bridge when we come to it I suppose.

Author: Resolving on behalf of @shartte (see https://discord.com/channels/313125603924639766/1154166981205962792/1406612774305665025)
